.github/workflows/pr-type-lint-spell.yml

```diff
@@ -0,0 +1,89 @@
+name: pr-type-lint-spell
+
+on:
+  pull_request:
+    branches: [ master ]
```

Comment:
Ok, running spell check and linter only on PR makes sense to me, as it will be the same after push (provided we only commits from PRs, which we do, as far as I know).
I'm not so sure about typing: it will be tested in PRs for Python 3.6 and 3.10, and after push on master for all Python versions. If there can be a difference for a specific Python version, why not check it on PRs? Because it would take too long?
Same for the unit tests. They are run for PRs in Python 3.6 - 3.9, for master push for Python 3.6 - 3.10, is there any reason for that? With the same logic, we could just skip the tests for push if they have been already run in the PR.
I'm not so sure about the medphys stuff, as there is little we can do if these fail, so maybe these can remain in master push only - undecided about that one.

Can you elaborate a bit on the strategy here? I understand that most (or all?) of this has already been this way before, but maybe we can reconsider this a bit while at it.

Reply:
* Typing on merge: Mostly I just wanted to add the type-hints badge to show typing is supported and passing. I can drop it in favour of adding more Python versions to the PR workflow, though. They don't take long.
* PR: The "core" tests run on 3.6 to 3.9, then a "core + extras (pixel data stuff)" run on 3.10, which mirrored the previous workflow. Alternatively I could run:
  * "core" on 3.6-3.10 and run a separate "extras" only on 3.10 which should be a bit faster.
  * "core + extras" on 3.6-3.10 which might take about the same time, depending on how slow the windows runs are

I tend to see the merge workflows as the comprehensive test that the `master` branch is in a good state (unit tests and type hints), and PR workflows should be a quicker test that the PR is good, but I'm open to suggestions. At the very least I think we should be testing more on Windows and MacOS (as the GDCM issues have shown).

I don't like have the pymedphys checks in the merge workflows since it makes it look like our checks are failing (and also why don't they just run their checks against pydicom dev?). At least by splitting it out its more visible to us that the failure isn't ours.

Reply:
I agree about the pymedphys checks, I'm actually not sure why they are running in pydicom instead of pymedphys itself, don't remember the history.

About the PR vs master workflow: I tend to see the PR workflow as the check that the code is ready to merge - so I like to have all needed tests there - and the master workflow more as a reference plus housekeeping stuff like pushing documentation and releases. If an issue shows up only in the tests on master, it is a bit inconvenient to fix it in a PR where it does not occur, especially if it occurs in an environment that you can't reproduce locally (say MacOS for me). 
But I also agree that a faster feedback from the CI is a good thing, so I have no strong opinion here.

Reply:
> I tend to see the PR workflow as the check that the code is ready to merge - so I like to have all needed tests there

I agree with the sentiment, but my take is that an error caught only in master merge (as we have them now) is so rare it is not worth increasing the time for CI in the PR.  It is only in platform-specific things like GDCM really where any trouble could reasonably happen.  If a PR was working on something for those, we could temporarily change the PR workflow to run extra tests.

As to pymedphys, perhaps @SimonBiggs can comment on whether there is something gained by running those tests in pydicom vs. running them in pymedphys against the pydicom master branch.  It seems so far that errors happening are not usually due to breaking changes in pydicom itself.

Reply:
> but my take is that an error caught only in master merge (as we have them now) is so rare it is not worth increasing the time for CI in the PR

Ok, makes sense. It is true that it happened very rarely so far, so it really isn't worth it.